src/Ormolu/Printer/Meat/Common.hs

```diff
@@ -62,15 +64,27 @@ p_ieWrappedName = \case
 -- | Render a @'LocatedN' 'RdrName'@.
 p_rdrName :: LocatedN RdrName -> R ()
 p_rdrName l = located l $ \x -> do
+  unboxedSums <- isExtensionEnabled UnboxedSums
   let wrapper = \case
         EpAnn {anns} -> case anns of
           NameAnnQuote {nann_quoted} -> tickPrefix . wrapper (ann nann_quoted)
-          NameAnn {nann_adornment = NameParens} -> parens N
+          NameAnn {nann_adornment = NameParens} ->
+            parens N . handleUnboxedSumsAndHashInteraction
           NameAnn {nann_adornment = NameBackquotes} -> backticks
           -- special case for unboxed unit tuples
           NameAnnOnly {nann_adornment = NameParensHash} -> const $ txt "(# #)"
           _ -> id
         EpAnnNotUsed -> id
+
+      -- When UnboxedSums is enabled, `(#` is a single lexeme, so we have to
+      -- insert spaces when we have a parenthesized operator starting with `#`.
+      handleUnboxedSumsAndHashInteraction
+        | unboxedSums,
+          -- Qualified names do not start wth a `#`.
+          Unqual (occNameString -> '#' : _) <- x =
```

Comment:
I am not 100% sure if the `Orig` or `Exact` constructors of [`RdrName`](https://hackage.haskell.org/package/ghc-lib-parser-9.6.2.20230523/docs/GHC-Types-Name-Reader.html#t:RdrName) could ever be relevant here :thinking: